utils/firestore.js

```diff
@@ -780,32 +780,186 @@ const retrieveSiteNotifications = async (siteId, isParent) => {
+        // If only firstName or lastName is in the query (no phone or email), only one array-contains operation is needed BUT we still need to handle the string and array cases for participantDoc.query.firstName and participantDoc.query.lastName
+        // If compound queries are executed, we need to handle results including duplicates and the mismatches caused by executing multiple queries
+        // If neither firstName nor lastName are in the query, we can run a simple query that doesn't need post-processing. This is the else statement. Return early since no additional processing is needed.
+        if (queries.firstName && queries.lastName && (queries.email || queries.phone)) {
+            const fNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.lastName, queries.phone, queries.email]);
+            const lNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.phone, queries.email]);
+            const phoneOrEmailExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.lastName]);
+            await handleQueries(fNameExtractedQuery, lNameExtractedQuery, phoneOrEmailExtractedQuery);
+        } else if (queries.firstName && queries.lastName) {
+            const fNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.lastName, queries.phone, queries.email]);
+            const lNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.phone, queries.email]);
+            await handleQueries(fNameExtractedQuery, lNameExtractedQuery, null);
+        } else if (queries.firstName && (queries.email || queries.phone)) {
+            const fNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.lastName, queries.phone, queries.email]);
+            const phoneOrEmailExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.lastName]);
+            await handleQueries(fNameExtractedQuery, null, phoneOrEmailExtractedQuery);
+        } else if (queries.lastName && (queries.email || queries.phone)) {
+            const lNameExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.phone, queries.email]);
+            const phoneOrEmailExtractedQuery = updateQueryForArrayContainsConstraints([queries.firstName, queries.lastName]);
+            await handleQueries(null, lNameExtractedQuery, phoneOrEmailExtractedQuery);
+        } else if (queries.firstName) {
+            await handleQueries(queries, null, null);
+        } else if (queries.lastName) {
+            await handleQueries(null, queries, null);
+        } else {
+            const nonNameQuery = generateQuery(collection, queries);
+            await executeQuery(nonNameQuery);
+
+            return fetchedResults;
         }
```

Comment:
Move this to a separate function

Reply:
I prefer this flow because it allows us to return early without additional execution or function calls. Let's leave this open for discussion in leads approval.